ddapm_test_agent/agent.py

```diff
@@ -333,50 +333,74 @@ async def _handle_traces(self, request: Request, version: Literal["v0.4", "v0.5"
         await checks.check("trace_stall", headers=request.headers, request=request)
 
         with CheckTrace.add_frame("headers") as f:
-            f.add_item(pprint.pformat(dict(request.headers)))
+            f.add_item(pprint.pformat(request.headers))
             await checks.check("meta_tracer_version_header", headers=request.headers)
             await checks.check("trace_content_length", headers=request.headers)
 
-            if version == "v0.4":
-                traces = self._decode_v04_traces(request)
-            elif version == "v0.5":
-                traces = self._decode_v05_traces(request)
-            log.info(
-                "received trace for token %r payload with %r trace chunks",
-                token,
-                len(traces),
-            )
-            for i, trace in enumerate(traces):
-                try:
-                    log.info(
-                        "Chunk %d\n%s",
-                        i,
-                        pprint_trace(trace, request.app["log_span_fmt"]),
-                    )
-                except ValueError:
-                    log.info("Chunk %d could not be displayed (might be incomplete).", i)
-            log.info("end of payload %s", "-" * 40)
-
-            with CheckTrace.add_frame(f"payload ({len(traces)} traces)"):
-                await checks.check(
-                    "trace_count_header",
-                    headers=request.headers,
-                    num_traces=len(traces),
+            try:
+                if version == "v0.4":
+                    traces = self._decode_v04_traces(request)
+                elif version == "v0.5":
+                    traces = self._decode_v05_traces(request)
+                log.info(
+                    "received trace for token %r payload with %r trace chunks",
+                    token,
+                    len(traces),
                 )
+                for i, trace in enumerate(traces):
+                    try:
+                        log.info(
+                            "Chunk %d\n%s",
+                            i,
+                            pprint_trace(trace, request.app["log_span_fmt"]),
+                        )
+                    except ValueError:
+                        log.info("Chunk %d could not be displayed (might be incomplete).", i)
+                log.info("end of payload %s", "-" * 40)
+
+                with CheckTrace.add_frame(f"payload ({len(traces)} traces)"):
+                    await checks.check(
+                        "trace_count_header",
+                        headers=request.headers,
+                        num_traces=len(traces),
+                    )
+            except Exception as e:
+                log.error(e)
 
         agent_url = request.app["agent_url"]
         if agent_url:
             log.info("Forwarding request to agent at %r", agent_url)
+
+            headers = {
```

Comment:
we shouldn't be using a `dict` for this as there can be duplicate http headers. To ensure we accurately pass the request along we should avoid removing duplicate headers.

Instead we should be using a list